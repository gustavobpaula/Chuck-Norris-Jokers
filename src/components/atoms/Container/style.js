import styled from 'styled-components';
import designSystem from '../../../designSystem';

export const Container = styled.div`
  margin: 0 auto;
  max-width: ${designSystem.get('display.sizes.monitor')};
  width: calc(100% - 30px);
`;
