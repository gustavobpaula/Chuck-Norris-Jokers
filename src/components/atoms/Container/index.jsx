import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './style';

const Container = ({ children }) => (
  <Styled.Container>{children}</Styled.Container>
);

Container.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Container;
