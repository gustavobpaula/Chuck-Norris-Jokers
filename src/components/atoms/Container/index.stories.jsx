import React from 'react';
import Container from './index';

export default {
  component: Container,
  title: 'Atoms/Container',
};

export const element = () => (
  <Container>
    <p>
      Lightning doesn&apos;t struck twice, Chuck Norris does. When Chuck Norris
      sends in his taxes, he sends blank forms and includes only a picture of
      himself, crouched and ready to attack. Chuck Norris has not had to pay
      taxes, ever, The Great Wall of China was originally created to keep Chuck
      Norris out. It failed miserably, Chuck Norris doesn&apos;t read books. He
      stares them down until he gets the information he wants, When Chuck Norris
      sends in his taxes, he sends blank forms and includes only a picture of
      himself, crouched and ready to attack. Chuck Norris has not had to pay
      taxes, ever. Fool me once, shame on you. Fool Chuck Norris once and he
      will roundhouse you in the face. What was going through the minds of all
      of Chuck Norris&apos; victims before they died? His shoe. Chuck Norris can
      slam a revolving door, Chuck Norris is so fast, he can run around the
      world and punch himself in the back of the head Chuck Norris doesn&apos;t
      go hunting... CHUCK NORRIS GOES KILLING. Chuck Norris drives an ice cream
      truck covered in human skulls, There is no chin behind Chuck Norris&apos;
      beard. There is only another fist. Chuck Norris has two speeds. Walk, and
      Kill.
    </p>
    <br />
    <p>
      Chuck Norris can see John Cena Chuck Norris doesn&apos;t wash his clothes,
      he disembowels them. Chuck Norris is the reason why Waldo is hiding. Chuck
      Norris is the only man to ever defeat a brick wall in a game of tennis.
    </p>
    <br />
    <p>
      Contrary to popular belief, Chuck Norris, not the box jellyfish of
      northern Australia, is the most venomous creature on earth, Chuck Norris
      doesn&apos;t churn butter. He roundhouse kicks the cows and the butter
      comes straight out, Chuck Norris originally appeared in the &quot;Street
      Fighter II&quot; video game, but was removed by Beta Testers because every
      button caused him to do a roundhouse kick. When asked bout this
      &quot;glitch,&quot; Norris replied, &quot;That&apos;s no glitch.&quot;,
      Chuck Norris doesn&apos;t churn butter. He roundhouse kicks the cows and
      the butter comes straight out Police label anyone attacking Chuck Norris
      as a Code 45-11... a suicide Chuck Norris made a bet with Hulk, whoever
      loses an arm wrestling would paint himself of green. The Pope kissed Chuck
      Norris&apos; ring Crop circles are Chuck Norris&apos; way of telling the
      world that sometimes corn needs to lie down, Chuck Norris doesn&apos;t
      read books. He stares them down until he gets the information he wants. If
      you ask Chuck Norris what time it is, he always says, &quot;Two seconds
      &apos;til.&quot; After you ask, &quot;Two seconds &apos;til what?&quot; he
      roundhouse kicks you in the face. Contrary to popular belief, America is
      not a democracy, it is a Chucktatorship. There is no theory of evolution.
      Just a list of animals Chuck Norris allows to live.
    </p>
    <br />
    <p>
      There is no theory of evolution. Just a list of creatures Chuck Norris has
      allowed to live. Police label anyone attacking Chuck Norris as a Code
      45-11... a suicide Chuck Norris doesn&apos;t churn butter. He roundhouse
      kicks the cows and the butter comes straight out. When the Boogeyman goes
      to sleep every night, he checks his closet for Chuck Norris Chuck Norris
      doesn&apos;t wear a watch. HE decides what time it is, Chuck Norris once
      roundhouse kicked someone so hard that his foot broke the speed of light,
      went back in time, and killed Amelia Earhart while she was flying over the
      Pacific Ocean Contrary to popular belief, America is not a democracy, it
      is a Chucktatorship Chuck Norris doesn&apos;t go hunting... CHUCK NORRIS
      GOES KILLING. Chuck Norris does not sleep. He waits. Chuck Norris
      originally appeared in the &quot;Street Fighter II&quot; video game, but
      was removed by Beta Testers because every button caused him to do a
      roundhouse kick. When asked bout this &quot;glitch,&quot; Norris replied,
      &quot;That&apos;s no glitch.&quot;.
    </p>
  </Container>
);
element.story = {
  name: 'Default',
};
