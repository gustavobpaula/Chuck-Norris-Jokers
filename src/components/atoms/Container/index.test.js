import React from 'react';
import { mount } from 'enzyme';

import Container from './index';

describe('<Container />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(
      <Container>
        <p>
          Lightning doesn&apos;t struck twice, Chuck Norris does. When Chuck
          Norris sends in his taxes, he sends blank forms and includes only a
          picture of himself, crouched and ready to attack. Chuck Norris has not
          had to pay taxes, ever, The Great Wall of China was originally created
          to keep Chuck Norris out. It failed miserably, Chuck Norris
          doesn&apos;t read books. He stares them down until he gets the
          information he wants, When Chuck Norris sends in his taxes, he sends
          blank forms and includes only a picture of himself, crouched and ready
          to attack. Chuck Norris has not had to pay taxes, ever. Fool me once,
          shame on you. Fool Chuck Norris once and he will roundhouse you in the
          face. What was going through the minds of all of Chuck Norris&apos;
          victims before they died? His shoe. Chuck Norris can slam a revolving
          door, Chuck Norris is so fast, he can run around the world and punch
          himself in the back of the head Chuck Norris doesn&apos;t go
          hunting... CHUCK NORRIS GOES KILLING. Chuck Norris drives an ice cream
          truck covered in human skulls, There is no chin behind Chuck
          Norris&apos; beard. There is only another fist. Chuck Norris has two
          speeds. Walk, and Kill.
        </p>
      </Container>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
