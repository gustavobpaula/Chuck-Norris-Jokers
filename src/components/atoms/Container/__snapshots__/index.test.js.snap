// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should mount properly 1`] = `
<Container>
  <style__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "style__Container-ce4g5i-0",
            "isStatic": false,
            "lastClassName": "cbwKLb",
            "rules": Array [
              "margin:0 auto;max-width:",
              "1200px",
              ";width:calc(100% - 30px);",
            ],
          },
          "displayName": "style__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "style__Container-ce4g5i-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="style__Container-ce4g5i-0 cbwKLb"
      >
        <p>
          Lightning doesn't struck twice, Chuck Norris does. When Chuck Norris sends in his taxes, he sends blank forms and includes only a picture of himself, crouched and ready to attack. Chuck Norris has not had to pay taxes, ever, The Great Wall of China was originally created to keep Chuck Norris out. It failed miserably, Chuck Norris doesn't read books. He stares them down until he gets the information he wants, When Chuck Norris sends in his taxes, he sends blank forms and includes only a picture of himself, crouched and ready to attack. Chuck Norris has not had to pay taxes, ever. Fool me once, shame on you. Fool Chuck Norris once and he will roundhouse you in the face. What was going through the minds of all of Chuck Norris' victims before they died? His shoe. Chuck Norris can slam a revolving door, Chuck Norris is so fast, he can run around the world and punch himself in the back of the head Chuck Norris doesn't go hunting... CHUCK NORRIS GOES KILLING. Chuck Norris drives an ice cream truck covered in human skulls, There is no chin behind Chuck Norris' beard. There is only another fist. Chuck Norris has two speeds. Walk, and Kill.
        </p>
      </div>
    </StyledComponent>
  </style__Container>
</Container>
`;
